### PR TITLE
CFY-2880 port aws manager blueprint to 3.3

### PR DESCRIPTION
**Pull request by ChenRoth**
create a aws ec2 manager blueprint that works with the new manager (3.3)

### CODE REVIEW

File: new/aws-ec2-manager-blueprint.yaml [hunk 1]
```diff
@@ -0,0 +1,675 @@
+tosca_definitions_version: cloudify_dsl_1_2
+
+imports:
+  - http://www.getcloudify.org/spec/cloudify/3.3m5/types.yaml
```
Comment (nir0s) [inline new/aws-ec2-manager-blueprint.yaml]: this should be m6 (m7 really.. but we don't have it right now)

File: new/aws-ec2-manager-blueprint.yaml [hunk 2]
```diff
@@ -0,0 +1,675 @@
...
+
+        policy_engine:
+          start_timeout: 30
+
+    interfaces:
+      cloudify.interfaces.lifecycle:
+        configure:
+          implementation: fabric.fabric_plugin.tasks.run_task
+          inputs:
+            tasks_file: components/manager/scripts/aws-ec2/configure.py
```
Comment (ChenRoth) [inline new/aws-ec2-manager-blueprint.yaml]: please do the same for openstack.

File: new/aws-ec2-manager-blueprint.yaml [hunk 3]
```diff
@@ -0,0 +1,675 @@
...
+            task_name: configure_manager            
+            task_properties:
+              aws_config: { get_property: [ aws_configuration, aws_config ] }
+              manager_config_path: { get_property: [SELF, cloudify, plugins, aws_config_path] }
+
+            fabric_env:
```
Comment (ChenRoth) [inline new/aws-ec2-manager-blueprint.yaml]:
```yaml
fabric_env: &my_fabric_env
    user: ...
    key_filename: ....



...


fabric_env: *my_fabric_env
```

File: new/aws-ec2-manager-blueprint.yaml [hunk 4]
```diff
@@ -0,0 +1,675 @@
...
+        target: nginx
+        target_interfaces:
+          cloudify.interfaces.relationship_lifecycle:
+            postconfigure: components/manager/scripts/set_manager_private_ip.sh
+
+
+  management_keypair:
+    type: cloudify.aws.nodes.KeyPair
+    properties:
+      use_external_resource: { get_input: use_existing_manager_keypair }
```
Comment (ChenRoth) [inline new/aws-ec2-manager-blueprint.yaml]: I think you can use the same referencing mechanism as in fabric_env while overriding the differentiating field(s)

File: new/components/manager/scripts/aws-ec2/configure.py [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+
+    if aws_config.get('ec2_region_name'):
+        region = get_region(aws_config['ec2_region_name'])
+        credentials.add_section('Boto')
+        credentials.set('Boto', 'ec2_region_name',
+                        aws_config['ec2_region_name'])
+        credentials.set('Boto', 'ec2_region_endpoint',
+                        region.endpoint)
+
+    credentials_string = StringIO()
```
Comment (ChenRoth) [inline new/components/manager/scripts/aws-ec2/configure.py]: unneeded..

File: new/components/manager/scripts/aws-ec2/configure.py [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+    credentials.write(credentials_string)
+
+    with open(temp_config, 'w') as temp_config_file:
+        temp_config_file.write(credentials_string.getvalue())
+
+    fabric.api.sudo('mkdir -p {0}'.
+                    format(os.path.dirname(manager_config_path)))
+    fabric.api.put(temp_config,
+                   '{0}'.format(manager_config_path),
```
Comment (ChenRoth) [inline new/components/manager/scripts/aws-ec2/configure.py]: remove string formatting..

File: new/aws-ec2-manager-blueprint-inputs.yaml [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+use_existing_manager_group: false
+use_existing_agent_group: false
+use_existing_manager_keypair: false
+use_existing_agent_keypair: false
+ssh_key_filename: ~/.ssh/cloudify-manager-kp.pem
+agent_private_key_path: ~/.ssh/cloudify-agent-kp.pem
+mananger_security_group_name: cloudify-manager-security-group
+agent_security_group_name: cloudify-agent-security-group
+manager_server_name: cloudify-manager-server
+manager_server_user: ubuntu
```
Comment (ChenRoth) [inline new/aws-ec2-manager-blueprint-inputs.yaml]: change to 'centos'

**Pull request merged**